Do not assign id to anonymous customers

SunnyYu-98 opened the pull request:
**A Brief Overview**
Fixes https://github.com/BroadleafCommerce/QA/issues/4305 by not assigning `customerId` until persisting the `Customer`

SunnyYu-98 committed:
Assign CustomerId when Customer is being persisted

- Allow null customerId to be set when creating a new Customer
- Add set customerId logic before persisting it to database
- Move saveCustomer logic from OrderDao to OrderService, and only save customer if customer is anonymous (if it's a logged in user, customer will already be persisted)
- Move setUsername logic from OrderDao to CustomerService#saveCustomer
- Remove outdated javadoc on passwordEncoder

SunnyYu-98 committed:
Update outdated javadoc method names

riteshadhikari17 commented:
@SunnyYu-98 There are Integration test failures. You might have to adjust the tests.

SunnyYu-98 commented:
@riteshadhikari17 Sounds good, thank you

SunnyYu-98 committed:
Save customer when creating a new cart in OrderService no matter what since we moved the saveCustomer logic to the CustomerService

- Update tests to generate customerId when creating a customer for the tests

SunnyYu-98 committed:
Generates customerId if passed in id is null and add new dedicate method to create Customer with null id

- Revert previous changes in CustomerService#createCustomerFromId to assign an id if given customerId is null
- Update initRegistrationForm and CustomerStateRequestProcessor to use the new createCustomerWithNullId for initializing customer registration form and creating an anonymous customer
- Deprecate CustomerService#createNewCustomer as it's no longer used and it's the exact same thing as CustomerService#createCustomer
- Revert previous change in CommonSetupBaseTest#createCustomer
- Update javadocs

SunnyYu-98 commented:
@riteshadhikari17 Sorry about the delay, got caught up with another project. I have updated the PR, all the tests seem to be passing now
